### PR TITLE
Focus order for slide-group, slide-panel, slide-panel-manager

### PR DESCRIPTION
**Pull request by artpark**
[UXPLATFORM-8606](https://jira2.cerner.com/browse/UXPLATFORM-8606)
===

### Summary
Adds the correct focus when navigating with keyboard controls. When disclosing the slide panel, focus will shift back to the element that caused the slide panel to open in the first place. This works with nested slide panels as well.

### Testing
Jest and WDIO tests (WDIO tests are currently failing and will be fixed with https://github.com/cerner/terra-framework/pull/1610)

### CODE REVIEW

File: packages/terra-slide-panel/src/SlidePanelUtils.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Returns true if the given element is focusable, otherwise returns false.
+ */
+const isFocusable = element => {
```
Comment (trandrew1023) [inline packages/terra-slide-panel/src/SlidePanelUtils.jsx:4]: Is there a case where the event.target could be null in the SlidePanel?

**Review by trandrew1023 — APPROVED** [review]

File: packages/terra-slide-panel/tests/wdio/__snapshots__/reference/clinical-lowlight-theme/en/chrome_large/slide-panel-spec/toggle_button_focused.png
[the remark this answers is not in the document]
Comment (artpark) [reply packages/terra-slide-panel/tests/wdio/__snapshots__/reference/clinical-lowlight-theme/en/chrome_large/slide-panel-spec/toggle_button_focused.png:1]: No reason, the existing test was already like that

**Review by sdadn — APPROVED** [review]

**Comment by eawww** [comment]
Looks good from the code. Still wanna give it a go when you have a deployment of it.

**Review by eawww — APPROVED** [review]

**Comment by artpark** [comment]
Removed PR site preview functionality commit (290de18) as it is already opened in https://github.com/cerner/terra-framework/pull/1618

**Pull request merged**